backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/addedvalue/interim/InterimContractLog.java

```diff
@@ -0,0 +1,171 @@
+import de.metas.contracts.modular.log.LogEntryCreateRequest;
+import de.metas.contracts.modular.log.LogEntryDocumentType;
+import de.metas.contracts.modular.workpackage.impl.AbstractPurchaseContractHandler;
+import de.metas.i18n.ExplainedOptional;
+import de.metas.i18n.IMsgBL;
+import de.metas.lang.SOTrx;
+import de.metas.money.Money;
+import de.metas.order.IOrderBL;
+import de.metas.order.OrderId;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.LocalDateAndOrgId;
+import de.metas.organization.OrgId;
+import de.metas.product.IProductBL;
+import de.metas.product.ProductId;
+import de.metas.product.ProductPrice;
+import de.metas.quantity.Quantity;
+import de.metas.quantity.Quantitys;
+import de.metas.uom.UomId;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.Getter;
+import lombok.NonNull;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.adempiere.warehouse.WarehouseId;
+import org.adempiere.warehouse.api.IWarehouseBL;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_M_Warehouse;
+import org.springframework.stereotype.Component;
+
+@Component
+@Getter
+public class InterimContractLog extends AbstractPurchaseContractHandler
+{
+
+	private final IOrderBL orderBL = Services.get(IOrderBL.class);
+
+	private final IOrgDAO orgDAO = Services.get(IOrgDAO.class);
+	private final IWarehouseBL warehouseBL = Services.get(IWarehouseBL.class);
+	private final IProductBL productBL = Services.get(IProductBL.class);
+	private final IFlatrateBL flatrateBL = Services.get(IFlatrateBL.class);
+	private final IMsgBL msgBL = Services.get(IMsgBL.class);
+
+	@NonNull private final ModCntrInvoicingGroupRepository modCntrInvoicingGroupRepository;
+
+	@NonNull private final AVInterimComputingMethod computingMethod;
+	@NonNull private final LogEntryDocumentType logEntryDocumentType = LogEntryDocumentType.CONTRACT_PREFINANCING;
+	@NonNull private final LogEntryContractType logEntryContractType = LogEntryContractType.MODULAR_CONTRACT;
+
+	public InterimContractLog(
+			@NonNull final ModularContractService modularContractService,
+			@NonNull final ModCntrInvoicingGroupRepository modCntrInvoicingGroupRepository,
+			@NonNull final AVInterimComputingMethod computingMethod)
+	{
+		super(modularContractService, modCntrInvoicingGroupRepository);
+		this.computingMethod = computingMethod;
+		this.modCntrInvoicingGroupRepository = modCntrInvoicingGroupRepository;
+	}
+
+	@Override
+	public @NonNull ExplainedOptional<LogEntryCreateRequest> createLogEntryCreateRequest(@NonNull final CreateLogRequest request)
+	{
+		final TableRecordReference tableRecordReference = request.getRecordRef();
+		final FlatrateTermId flatrateTermId = FlatrateTermId.ofRepoId(tableRecordReference.getRecordIdAssumingTableName(getSupportedTableName()));
+		final I_C_Flatrate_Term flatrateTermRecord = flatrateBL.getById(flatrateTermId);
+		final I_C_Flatrate_Term modularContractRecord;
```

Comment:
was this part of an if/else ? Looks weird being defined here but initialized below.